NetIO/MsgChannel.cpp

```diff
@@ -23,6 +23,13 @@
 #include <cstring>
 #include "errors.h"
 
+DS::MsgChannel::MsgChannel(bool initFd)
+    : m_semaphore(-1)
+{
+    if (initFd)
+        fd();
```

Comment:
Unfortunately, this moves the `SystemError` exception throw to the `MsgChannel` constructor, which is harder to catch.  However, considering the likelihood of the eventfd creation failing (without also seeing other issues on the server), I'm willing to allow it.

Reply:
I'm curious to know what aside from exceeding the maximum fd limit would cause that to happen. I'm sure it's an exciting situation fraught with all sorts of interesting problems 😄 (sarcasm self-test complete)

Reply:
More seriously, I was thinking about this a bit more last night. I remembered that there is a mutex lock in both the `putMessage()` and `getMessage()` methods. Perhaps the call to `fd()` should be moved during the mutex lock to make the exception handling saner?

Reply:
`fd()` is used outside of `getMessage()` and `putMessage()` though.  Perhaps `fd()` itself should use the mutex...

Reply:
That looks like a good solution to me